Content.Shared/NPC/Components/NpcTimedFactionMemberComponent.cs

```diff
@@ -0,0 +1,62 @@
+using Content.Shared.NPC.Prototypes;
+using Content.Shared.NPC.Systems;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+
+namespace Content.Shared.NPC.Components;
+
+/// <summary>
+/// On randomized timer, adds the wearer to a faction.
+/// On a second timer, removes the wearer from said faction.
+/// </summary>
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentPause, Access(typeof(NpcTimedFactionSystem))]
+public sealed partial class NpcTimedFactionComponent : Component
+{
+    /// <summary>
+    /// When the toggled faction will be added.
+    /// </summary>
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer))]
+    [AutoPausedField]
+    public TimeSpan TimeFactionChange = TimeSpan.Zero;
+
+    /// <summary>
+    /// When the toggled faction will be removed.
+    /// </summary>
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer))]
+    [AutoPausedField]
+    public TimeSpan TimeFactionChangeBack = TimeSpan.Zero;
+
```

Comment:
Can't you just have 1 timespan field for state change and a bool as to the current state?